### PR TITLE
libc: Add uuid implemenation

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Adds the UUID libc functions specified by OpenGroup here:
https://pubs.opengroup.org/onlinepubs/009629399/toc.htm

### CODE REVIEW

File: libs/libc/uuid/lib_uuid_from_string.c [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  /* Assume the worst. */
+
+  if (status != NULL)
+    {
+      *status = uuid_s_invalid_string_uuid;
+    }
+
+  /* The UUID string representation has a fixed length. */
+
+  if (strlen(s) != UUID_STR_LEN)
```
Comment (btashton) [inline libs/libc/uuid/lib_uuid_from_string.c:65]: Should we protect this with `strnlen(s, UUID_STR_LEN+1)`

File: libs/libc/uuid/lib_uuid_stream.c [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/types.h>
+#include <endian.h>
+#include <uuid.h>
+
+/****************************************************************************
```
Comment (btashton) [inline libs/libc/uuid/lib_uuid_stream.c:29]: Did this code come from NetBSD?  It looks a lot like the code here
http://mirrors.mit.edu/NetBSD/NetBSD-release-7/src/lib/libc/uuid/uuid_stream.c

If so my understanding is that we would leave the BSD headers on this and call it out in the LICENSE file

File: libs/libc/uuid/lib_uuid_stream.c [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+ *  |clk_seq_hi_res |  clk_seq_low  |         node (0-1)            |
+ *  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *  |                         node (2-5)                            |
+ *  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *
+ ****************************************************************************/
+
+/* Alignment-agnostic encode/decode bytestream to/from little/big endian. */
+
+static inline uint16_t be16dec(const void *pp)
```
Comment (btashton) [inline libs/libc/uuid/lib_uuid_stream.c:49]: These endian functions dont seem to be uuid specific.  Normally they are defined via `<machine/endian.h>` or `<endian.h>` OS specific location.

File: libs/libc/uuid/lib_uuid_from_string.c (hunk 1 above)
Comment (xiaoxiang781216) [reply libs/libc/uuid/lib_uuid_from_string.c:65]: Done.

File: libs/libc/uuid/lib_uuid_stream.c (hunk 3 above)
Comment (xiaoxiang781216) [reply libs/libc/uuid/lib_uuid_stream.c:49]: Done.

File: libs/libc/uuid/lib_uuid_stream.c (hunk 2 above)
Comment (xiaoxiang781216) [reply libs/libc/uuid/lib_uuid_stream.c:29]: The code is rewritten from scratch now. Should we put some note in LICENSE file?

Comment (btashton) [reply libs/libc/uuid/lib_uuid_stream.c:29]: I think this is fine now.  Most of these functions are trivial especially with us using our existing endian code.

**Pull request merged**